src/audio/module_adapter/module_adapter.c

```diff
@@ -792,19 +772,15 @@ static int module_adapter_simple_copy(struct comp_dev *dev)
 	list_for_item(blist, &dev->bsource_list)
 		buffer_release(source_c[i++]);
 
-	for (i = 0; i < mod->num_output_buffers; i++)
-		mod->output_buffers[i].size = 0;
-
-	for (i = 0; i < mod->num_input_buffers; i++) {
-		mod->input_buffers[i].size = 0;
-		mod->input_buffers[i].consumed = 0;
-	}
-
 	return ret;
 }
 
 int module_adapter_copy(struct comp_dev *dev)
 {
+	struct input_stream_buffer input_buffers[PLATFORM_MAX_STREAMS];
+	struct output_stream_buffer output_buffers[PLATFORM_MAX_STREAMS];
```

Comment:
i need to put it somewhere ;) if we don't want it pre-alocated
But good question
in zephyr:
#define ZEPHYR_LL_STACK_SIZE	8192

what is in NXP? no idea

Reply:
On NXP we probably have a similar stack size, but our architecture gives us so much memory we can expand it easily if needed (i.MX8 uses shared memory from the host, upcoming i.MX93 has roughly 800MB [!!] of total memory allocated in the current Jailhouse proposal; it will possibly be reduced but still be way more than on Intel even after that reduction).